lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py

```diff
@@ -723,4 +723,80 @@ def test_check_geospatial_vertical_units(self):
 ################################################################################
 
 class TestIMOS1_4(TestIMOS1_3):
-    pass
+
+    def setUp(self):
+        '''
+        Initialize the dataset
+        '''
+        super(TestIMOS1_4, self).setUp()
+        self.imos = IMOS1_4Check()
+        # good_dataset was only all good for IMOS 1.3!
+        self.old_good_dataset = self.good_dataset
+
+    def test_check_mandatory_global_attributes(self):
+        attributes = set(['Conventions',
+                          'project',
+                          'naming_authority',
+                          'data_centre',
+                          'data_centre_email',
+                          'date_created',
+                          'title',
+                          'abstract',
+                          'author',
+                          'principal_investigator',
+                          'citation',
+                          'acknowledgement',
+                          'disclaimer',
+                          'license',
+                          'standard_name_vocabulary'])
+
+        att_changed = set(['Conventions',
+                           'data_centre',
+                           'data_centre_email',
+                           'disclaimer',
+                           'license',
+                           'standard_name_vocabulary'])
+
+        ret_val = self.imos.check_mandatory_global_attributes(self.new_dataset)
+        att_passed = set([r.name[1] for r in ret_val if r.value])
+        self.assertEqual(att_passed, attributes)
+
+        ret_val = self.imos.check_mandatory_global_attributes(self.bad_dataset)
+        att_failed = set([r.name[1] for r in ret_val if not r.value])
+        self.assertEqual(att_failed, attributes)
+
+        ret_val = self.imos.check_mandatory_global_attributes(self.old_good_dataset)
+        att_all = set([r.name[1] for r in ret_val])
+        att_failed = set([r.name[1] for r in ret_val if not r.value])
+        self.assertEqual(att_all, attributes)
```

Comment:
`att_all` is the set of all attributes checked by `check_mandatory_global_attributes` (whether the check passed or failed).
`attributes` is the set of all attributes that *should* have been checked.